staging/src/k8s.io/apiserver/pkg/authentication/request/bearertoken/bearertoken.go

```diff
@@ -39,8 +39,8 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 	if auth == "" {
 		return nil, false, nil
 	}
-	parts := strings.Split(auth, " ")
-	if len(parts) < 2 || strings.ToLower(parts[0]) != "bearer" {
+	parts := strings.SplitN(auth, " ", 3)
+	if len(parts) != 2 || strings.ToLower(parts[0]) != "bearer" {
```

Comment:
wouldn't this break a request that included a trailing space after the token?

Reply:
Yes, that's right.
But is there a use case to specify any values after token?

Reply:
I expect the main cause would be accidental header construction. If this were new code, being strict would be fine. Since there are existing clients that would be impacted, this change introduces unnecessary risk.

Reply:
Agreed.
So, would this be fine?
```diff
-     	parts := strings.Split(auth, " ")
+ 	parts := strings.SplitN(auth, " ", 3) 
```

I mean that I don't edit the bellow code and merely make Split() bit faster.
```
if len(parts) < 2 || strings.ToLower(parts[0]) != "bearer" {
```

Reply:
that would be fine

Reply:
Thank you!
I'll change the code.

Reply:
I modified the code and PR title.

Thank you for confirming!